packages/dom-expressions/src/client.js

```diff
@@ -84,22 +84,36 @@ export function addEventListener(node, name, handler, delegate) {
 }
 
 export function classList(node, value, prev = {}) {
-  const classKeys = Object.keys(value),
-    prevKeys = Object.keys(prev);
+  const prevKeys = Object.keys(prev);
   let i, len;
   for (i = 0, len = prevKeys.length; i < len; i++) {
     const key = prevKeys[i];
     if (!key || key === "undefined" || key in value) continue;
     toggleClassKey(node, key, false);
     delete prev[key];
   }
```

Comment:
Scenario 1, wipe all classes. These will be rebuilt in setClasses

```suggestion
  for (const key of Object.keys(prev)) {
    toggleClassKey(node, key, false);
    delete prev[key];
  }
```